Assets/Editor Toolbox/Editor/SceneView/ToolboxEditorSceneViewObjectSelector.cs

```diff
@@ -0,0 +1,344 @@
+﻿using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+namespace Toolbox.Editor.SceneView
+{
+    public class ToolboxEditorSceneViewObjectSelector : EditorWindow
+    {
+        private List<GameObject> gameObjects;
+        private List<string> gameObjectPaths;
+
+        private const float sizeXPadding = 2f;
+        private const float sizeYPadding = 2f;
+        private const float buttonYSpacing = 0.0f;
+        private const float buttonYSize = 20f;
+        private const float sizeXOffset = -30f;
+
+        private GameObject highlightedObject;
+        private Vector2 size;
+        private Vector2 buttonSize;
+        private GUIStyle buttonStyle;
+        private Color selectionColor = new Color(0.50f, 0.70f, 1.00f);
+
+        private int shiftMinSelectionID = -1;
+        private int shiftMaxSelectionID = -1;
+        private int shiftLastID = -1;
+
+        private GameObject HighlightedObject
+        {
+            set
+            {
+                if (highlightedObject == value)
+                {
+                    return;
+                }
+
+                highlightedObject = value;
+                UnityEditor.SceneView.RepaintAll();
+
+                if (highlightedObject != null)
+                {
+                    EditorGUIUtility.PingObject(highlightedObject);
+                }
+            }
+        }
+
+        public static void Show(List<GameObject> gameObjects, Vector2 position)
+        {
+            Rect rect = new Rect(position, Vector2.one);
+
+            ToolboxEditorSceneViewObjectSelector window = CreateInstance<ToolboxEditorSceneViewObjectSelector>();
+            window.wantsMouseMove = true;
+            window.wantsMouseEnterLeaveWindow = true;
+            window.gameObjects = gameObjects;
+            window.InitializeGameobjectPaths();
+            window.CalculateSize();
+            window.ShowAsDropDown(rect, window.size);
+        }
+
+        private void InitializeStyle()
+        {
+            buttonStyle = new GUIStyle(GUI.skin.button);
```

Comment:
I suggest using private static class Style to create and cache styles, e.g. you can ToolboxEditorHierarchy for an example